maker/maker.go

```diff
@@ -37,7 +43,35 @@ func GetReceiverTypeName(src []byte, fl interface{}) (string, *ast.FuncDecl) {
 		st = st[1:]
 	}
 	return st, fd
+}
+func GetStructName(src []byte, dec interface{}) string {
+
+	if gd, ok := dec.(*ast.GenDecl); ok {
```

Comment:
it is too many ifs inside ifs, for such type of construction is better to keep code a bit flatter as this:
```Go
gd, ok := dec.(*ast.GenDecl)
if !ok || gd.Tok != token.Type {
    return ""
}
ts, ok := gd.Specs[0].(*ast.TypeSpec)
if !ok {
    return ""
}
return ts.Name.Name
```

You code is correct, but having it a bit more narrow should generally be easier to interpret.